ArcGISRuntimeSDKQt_QMLSamples/Layers/RasterLayerFile/RasterLayerFile.qml

```diff
@@ -28,55 +27,67 @@ Rectangle {
 
     property real scaleFactor: System.displayScaleFactor
     property url dataPath: System.userHomePath + "/ArcGIS/Runtime/Data/raster/"
+    property var supportedFormats: ["img","I12","dt0","dt1","dt2","tc2","geotiff","tif", "tiff", "hr1","jpg","jpeg","jp2","ntf","png","i21","ovr"]
 
     MapView {
         id: mapView
         anchors.fill: parent
 
         Map {
-            Basemap {
-                RasterLayer {
-                    Raster {
-                        path: dataPath + "/Colorado.tif"
-                    }
-                }
+            id: map
+
+            BasemapImagery {}
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                createAndAddRasterLayer(dataPath + "Shasta.tif");
             }
         }
     }
 
-    Column {
-        anchors {
-            left: parent.left
-            top: parent.top
-            margins: 15
+    RasterLayer {
+        id: rasterLayer
+
+        Raster {
+            id: raster
         }
-        spacing: 10
 
-        Button {
-            text: "Add Raster"
-            width: 100 * scaleFactor
-            onClicked: fileDialog.open();
+        onLoadStatusChanged: {
+            if (loadStatus !== Enums.LoadStatusLoaded)
+                return;
+
+            mapView.setViewpointCenterAndScale(fullExtent.center, 80000);
         }
     }
 
-    FileDialog {
-        id: fileDialog
+    Button {
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            bottom: parent.bottom
+            margins: 24 * scaleFactor
+        }
+
+        text: "Add Raster"
+        width: 100 * scaleFactor
+        onClicked: loader.open();
+    }
 
-        // only display supported raster formats
-        nameFilters: ["Raster files (*.img *.I12 *.dt0 *.dt1 *.dt2 *.tc2 *.geotiff *.tif *.hr1 *.jpg *.jpeg *.jp2 *.ntf *.png *.i21 *.ovr)"]
+    RasterLoader {
+        id: loader
+        anchors.fill: rootRectangle
         folder: dataPath
+        supportedExtensions: supportedFormats
 
-        onAccepted: {
-            createAndAddRasterLayer(fileDialog.fileUrl)
+        onRasterFileChosen: {
+            createAndAddRasterLayer(url);
         }
     }
 
     function createAndAddRasterLayer(rasterUrl) {
-        var newRaster = ArcGISRuntimeEnvironment.createObject("Raster", {path: rasterUrl});
-        var newRasterLayer = ArcGISRuntimeEnvironment.createObject("RasterLayer", {raster: newRaster});
-        var newBasemap = ArcGISRuntimeEnvironment.createObject("Basemap");
-        newBasemap.baseLayers.append(newRasterLayer);
-        var newMap = ArcGISRuntimeEnvironment.createObject("Map", {basemap: newBasemap});
-        mapView.map = newMap;
+        map.operationalLayers.clear();
+        raster.path = rasterUrl;
```

Comment:
I wonder if we need to create a new RasterLayer each time. Or at the very least, create a new Raster each time. I'm not sure if you can continually change the path after it is loaded, and have everything continue to work as expected